osu.Game.Rulesets.Mania/Objects/Drawables/DrawableHoldNote.cs

```diff
@@ -7,14 +7,52 @@
 using OpenTK.Graphics;
 using osu.Framework.Configuration;
 using OpenTK.Input;
+using osu.Framework.Input;
+using OpenTK;
+using osu.Framework.Graphics.Containers;
+using osu.Game.Rulesets.Mania.Judgements;
+using osu.Framework.Extensions.IEnumerableExtensions;
 
 namespace osu.Game.Rulesets.Mania.Objects.Drawables
 {
+    /// <summary>
+    /// Visualises a <see cref="HoldNote"/> hit object.
+    /// </summary>
     public class DrawableHoldNote : DrawableManiaHitObject<HoldNote>
     {
-        private readonly NotePiece headPiece;
+        private readonly DrawableNote head;
+        private readonly DrawableNote tail;
+
         private readonly BodyPiece bodyPiece;
-        private readonly NotePiece tailPiece;
+        private readonly Container<DrawableHoldNoteTick> tickContainer;
+
+        /// <summary>
+        /// Time at which the user started holding this hold note.
+        /// </summary>
+        private double holdStartTime;
+
+        /// <summary>
+        /// Whether the hold note has been released too early and shouldn't give full score for the release.
+        /// </summary>
+        private bool hasBroken;
+
+        private bool _holding;

```

Comment:
rather than this backing field, how about making `holdStartTime` nullable?

it doesn't make sense for it to be non-nullable anyways, as there's no way to know if a user started holding at 0ms or not.